tests/functional/TicketTask.php

```diff
@@ -683,4 +689,373 @@ public function testParentMetaSearchOptions()
+
+        //Create profile
+        $profile = $this->createItem(Profile::class, [
+            'name' => 'testUpdateMyTask',
+            'interface' => 'central',
+        ]);
+
+        $profile_user = new Profile_User();
+        $profile_user->getFromDBByCrit([
+            'users_id' => $uid,
+        ]);
+
+        $DB->delete(
+            ProfileRight::getTable(),
+            [
+                'profiles_id' => $profile->getID(),
+                'name' => 'task'
+            ],
+        );
+
+        //Add the profile to the user
+        $DB->update(
+            Profile_User::getTable(),
+            ['profiles_id' => $profile->getID()],
+            ['users_id' => $uid],
+        );
+
+        $this->login();
+
+        $this->boolean((new \TicketTask())->canCreate())->isFalse();
+
+        $ticket_right = new ProfileRight();
+        $ticket_right->getFromDBByCrit(
+            [
+                'profiles_id' => $profile->getID(),
+                'name' => 'ticket',
+            ]
+        );
+
+        // Create a profile right to be able to read my ticket
+        $this->updateItem(
+            ProfileRight::class,
+            $ticket_right->getID(),
+            [
+                'rights' => READ
+            ]
+        );
+
+        // Create a profile right to add a task to the ticket
+        $task_right = $this->createItem(ProfileRight::class, [
+            'profiles_id' => $profile->getID(),
+            'name' => 'task',
+            'rights' => CommonITILTask::ADDMY,
+        ]);
+
+        $this->login();
+
+        $this->boolean((new \TicketTask())->canCreate())->isTrue();
+
+        // Check that the last_rights_update field is not null
+        $this->variable($profile->fields['last_rights_update'])->isNotNull();
+
+        $profile->getFromDB($profile->getID());
+
+        $this->login();
+
+        // Get a ticket
+        $ticket = self::getNewTicket();
+
+        // Create a task
+        $task = $this->createItem(
+            \TicketTask::class,
+            [
+                'state' => \Planning::TODO,
+                'tickets_id' => $ticket,
+                'content' => "Task",
+                'users_id' => $uid,
+            ]
+        );
+
+        $this->boolean((new \TicketTask())->canUpdateItem())->isFalse();
+
+        // Create a profile right to add a task to the ticket
+        $this->updateItem(
+            ProfileRight::class,
+            $task_right->getID(),
+            [
+                'rights' => CommonITILTask::UPDATEMY,
+            ]
+        );
+
+        $this->login();
+
+        $this->boolean($task->canUpdateItem())->isTrue();
+    }
+
+    /**
+     * Test that the user can create a task in function of his status
+     *
+     * @param string $profileName
+     * @param integer $right
+     * @param string $ticketUserType
+     *
+     * @return void
+     */
+    private function createTaskAndAssertCanCreate($profileName, $right, $ticketUserType)
```

Comment:
Please use strict signatures for new methods.

```suggestion
    private function createTaskAndAssertCanCreate(string $profileName, string $right, string $ticketUserType): void
```